pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -228,6 +228,11 @@ static PyObject *get_values(PyObject *obj) {
     PRINTMARK();
 
     if (values && !PyArray_CheckExact(values)) {
+
+        if (PyObject_HasAttrString(values, "to_numpy")) {
```

Comment:
can you add a comment here about what this is doing, and . TODO to see if we can remove the .values below